### DISCUSSION TITLE
Error encontered with addDoubletScores() function

### DISCUSSION THREAD

**Discussion by sylestiel** [GreenleafLab/ArchR#1044] [General]
Hi, 

 I upgraded to R version 4.1.1 and encountered an error  at the addDoubletScores(proj) step. 
<img width="943" alt="Screen Shot 2021-09-22 at 5 06 22 PM" src="https://user-images.githubusercontent.com/64283689/134428692-81eb66db-c00f-4fc1-9867-8caedaacbc3b.png">

[ArchR-addDoubletScores-4e4866ee3f3d-Date-2021-09-22_Time-16-46-41.log](https://github.com/GreenleafLab/ArchR/files/7214101/ArchR-addDoubletScores-4e4866ee3f3d-Date-2021-09-22_Time-16-46-41.log)

This has not been an issue with earlier R version.

Can you help throw some light on this problem.

Thanks!

**Comment by rcorces** [comment]
https://github.com/GreenleafLab/ArchR/discussions/904#discussioncomment-1320628
Some sort of compatibility error. We dont have bandwidth to address this right now so just avoid R 4.1.1

**Reply by sylestiel** [reply]
I installed and read in the following to be able to run ArchR using R 4.1.1
library(parallel)
library(magick)

Appears to be running now.

**Comment by jenellewallace** [comment]
Just to let you know, I'm using R version 4.1.0 (2021-05-18) on Ubuntu 20.04.3 LTS and I also found it necessary to run "library(parallel") to get multiple threads to work.

**Comment by rcorces** [comment]
Re: `parallel` in newer versions of R

This had been bothering me for a while so I finally tracked it down. Description below.

In a recent update in Bioconductor, the `BiocGenerics` function stopped depending on and importing `parallel`. So in newer versions of Bioconductor, `parallel` no longer gets loaded alongside `BiocGenerics` and thus needed to be loaded separately. I've fixed this by adding a require statement when `mclapply` runs: https://github.com/GreenleafLab/ArchR/commit/0630ae09571734338520a16f801cc8a71a06d2f1

This has been fixed on `release_1.0.2`

see https://bioconductor.org/packages/release/bioc/html/BiocGenerics.html
vs 
https://bioconductor.org/packages/3.1/bioc/html/BiocGenerics.html
